internal/pkg/config/output.go

```diff
@@ -120,6 +120,12 @@ func (c *Elasticsearch) ToESConfig() (elasticsearch.Config, error) {
 		h.Set(key, val)
 	}
 
+	// Set special header "X-elastic-product-origin" for .fleet-* indices based on the latest conversation with ES team
+	// This eliminates the warning while accessing the system index
+	c.Headers = map[string]string{
+		"X-elastic-product-origin": "fleet",
+	}
```

Comment:
You need to change this to:

```
h.Set("X-elastic-product-origin", "fleet")
```

Modifying the `c.Headers` is not something `ToESConfig` should do.

Reply:
🤦